core/chains/evm/types/models.go

```diff
@@ -118,16 +118,23 @@ func (h *Head) IsInChain(blockHash common.Hash) bool {
 // HashAtHeight returns the hash of the block at the given height, if it is in the chain.
 // If not in chain, returns the zero hash
 func (h *Head) HashAtHeight(blockNum int64) common.Hash {
-	for {
+	headAtHeight, err := h.HeadAtHeight(blockNum)
+	if err != nil {
+		return common.Hash{}
+	}
+
+	return headAtHeight.BlockHash()
+}
+
+func (h *Head) HeadAtHeight(blockNum int64) (commontypes.Head[common.Hash], error) {
```

Comment:
I'm not feeling great about this as we're not staying consistent with the existing method. `HashAtHeight` can return  an empty hash, but `HeadAtHeight` returns an error instead.